### PR TITLE
feat(config): implement settings and labels management

### PR DESCRIPTION
**Pull request by flexiondotorg**
## Summary

Complete implementation of repository settings and labels management for tailor. Adds read/write paths for repository settings (branch protection, security, workflow configurations), comprehensive label management with Catppuccin Latte defaults, and validation/synchronisation logic for configuration consistency.

## Changes

- Repository settings: implement read/write operations for extended GitHub repository configuration
- Label management: full lifecycle support with name casing synchronisation and comprehensive test coverage
- Configuration validation: load-time validation of repository settings and label specifications
- Specification documentation: detailed implementation details and manual sponsorships step
- Test coverage: 1800+ lines of test code covering settings, labels, and configuration validation

## Testing

- Unit tests across all modules: settings, labels, configuration validation and write operations
- Integration tests verify settings read/write and label synchronisation
- Configuration tests validate label merging, YAML structure, and edge cases
- Manual testing of label casing synchronisation and swatch configuration

## Related Issues

Resolves implementation of complete settings and labels feature with two code review iterations and specification documentation.

### CODE REVIEW

File: internal/alter/alter.go [hunk 1]
```diff
@@ -38,7 +38,9 @@ func Run(cfg *config.Config, dir string, mode ApplyMode, client *api.RESTClient)
 	// the caller requested a recut.
 	if shouldMerge(cfg, mode) {
 		added := config.MergeDefaultSwatches(cfg)
-		if len(added) > 0 && mode.ShouldWrite() {
+		repoMerged := config.MergeDefaultRepoSettings(cfg)
+		labelsMerged := config.MergeDefaultLabels(cfg)
```
[the remark this answers is not in the document]
Comment (flexiondotorg) [reply internal/alter/alter.go:42]: False positive. The merge step runs inside shouldMerge() gating and the behaviour is explicitly specified - populating missing sections with defaults is the intended design. Downstream processors guard against nil/empty correctly after merge.
